src/dataregistry/registrar/dataset.py

```diff
@@ -270,203 +412,163 @@ def _register_or_replace(
+              All other properties are what the user specifies in the replace
+              function
+            - The old dataset gets pointed to the new dataset saying it is the
+              most up to date itteration
 
-        # Pull the dataset properties together
-        values = {"name": name, "relative_path": relative_path}
-        values["version_major"] = v_fields["major"]
-        values["version_minor"] = v_fields["minor"]
-        values["version_patch"] = v_fields["patch"]
-        values["version_string"] = version_string
-        if version_suffix:
-            values["version_suffix"] = version_suffix
-        if description:
-            values["description"] = description
-        if execution_id:
-            values["execution_id"] = execution_id
-        if access_api:
-            values["access_api"] = access_api
-        if access_api_configuration:
-            values["access_api_configuration"] = _read_configuration_file(
-                access_api_configuration, max_config_length
-            )
-        values["is_overwritable"] = is_overwritable
-        values["is_overwritten"] = False
-        values["is_archived"] = False
-        values["register_date"] = datetime.now()
-        values["owner_type"] = owner_type
-        values["owner"] = owner
-        values["creator_uid"] = self._uid
-        values["register_root_dir"] = self._root_dir
-        values["location_type"] = location_type
-        if url and location_type == "external":
-            values["url"] = url
-        if contact_email:
-            values["contact_email"] = contact_email
-        values["replace_iteration"] = replace_iteration
+        Returns
+        -------
+        prim_key : int
+            The dataset ID of the new row relating to this entry (else None)
+        execution_id : int
+            The execution ID associated with the dataset
+        """
 
-        # We tentatively start with an "invalid" dataset in the database. This
-        # will be upgraded to valid if the data copying (if any) was successful.
-        values["status"] = 0
+        # Validate the inputs we are working with
+        self._validate_register_inputs(name, version, kwargs_dict)
 
-        # Create a new row in the data registry database.
-        with self._engine.connect() as conn:
-            prim_key = add_table_row(conn, dataset_table, values, commit=True)
+        # Compute version string
+        self._compute_version_string(name, version, kwargs_dict)
 
-        # Get dataset characteristics; copy to `root_dir` if requested
-        if location_type == "dataregistry":
-            (
-                dataset_organization,
-                num_files,
-                total_size,
-                ds_creation_date,
-            ) = self._handle_data(
-                relative_path, old_location, owner, owner_type, verbose
+        # If `relative_path` not passed, automatically generate it
```

Comment:
Shouldn't there also be a check when `relative_path` is passed in that it matches the expected value?

Reply:
Ah this check shouldn't actually be needed at all, i've removed it. The relative path used is the same as the previous dataset (returned in found_previous), so it always gets overwritten in the code further down.

Reply:
Ok, that seems right.  But then it shouldn't even be an argument. Please take it out of the argument list.